### PR TITLE
Create mean_of_columns.d

### PR DESCRIPTION
**Pull request by JackStouffer**
ping @9il

### CODE REVIEW

**Comment by JackStouffer** [comment]
Updated

**Comment by JackStouffer** [comment]
Added Python benchmarking code

File: mean_of_columns.d [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+*
+*     import numpy
+*     data = numpy.arange(100000).reshape((100, 1000))
+*     means = numpy.mean(data, axis=0)
+*
+* and we benchmark the numpy.mean line using the following Python command,
+* 
+*    python -m timeit \
+*         -s 'import numpy; data = numpy.arange(100000).reshape((100, 1000))' \
+*         'means = numpy.mean(data, axis=0)'
```
Comment (9il) [inline mean_of_columns.d:17]: Unfair. Only `means = numpy.mean(data, axis=0)` should be tested.

Comment (JackStouffer) [inline mean_of_columns.d:17]: It is, `-s` marks one time setup code.

Comment (9il) [inline mean_of_columns.d:17]: Oh, I see. Thanks!

File: mean_of_columns.d [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        .sliced(100, 1000)
+        .transposed
+        .map!(r => sum(r) / cast(double) r.length)
+        .array;
+}
+
+void main() {
+    data = 100_000.iota.array;
+    auto r = benchmark!(f0)(10_000);
+    auto f0Result = to!Duration(r[0]);
```
Comment (9il) [inline mean_of_columns.d]: The last one

``` D
    enum testCount = 10_000;
    auto r = benchmark!(f0)(testCount);
    auto f0Result = to!Duration(r[0] / testCount);
```

Comment (JackStouffer) [inline mean_of_columns.d]: Updated. I tried doing that originally, but I was accidentally doing `auto f0Result = to!Duration(r[0]) / testCount;` instead and getting completely wrong results. Thanks for this fix.

**Comment by JackStouffer** [comment]
Thanks!

**Pull request merged**